app/services/notifications_list.rb

```diff
@@ -1,9 +1,20 @@
 class NotificationsList
-  def self.for(application_choice, include_ratifying_provider: false)
-    return application_choice.provider.provider_users.where(send_notifications: true) if application_choice.accredited_provider.nil? || !include_ratifying_provider
+  attr_reader :event
 
-    application_choice.provider.provider_users.where(send_notifications: true).or(
-      application_choice.accredited_provider.provider_users.where(send_notifications: true),
-    )
+  def self.for(application_choice, include_ratifying_provider: false, event: nil)
+    if FeatureFlag.active?(:configurable_provider_notifications)
+      raise 'Undefined type of notification event' if event.nil? || !ProviderUserNotification.method_defined?(event)
```

Comment:
This binds us 1:1 event->notification type — is that ok? (if so, probably no need to change this for an explicit mapping at this stage)

Reply:
I think `notification_type` may be a better name for this arg. 
For users with `application_received: true` we send up to 2 emails (application received and reminder email to the training providers), similarly for `offer_declined: true` there are 2 possible events (offer declined by the candidate and offer decline by default).

Reply:
Sure. Do we want to map in here or have the callers effectively do the mapping when they pass the `notification_type`?